src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -122,6 +136,34 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         _KeyBindingList = single_threaded_observable_vector(std::move(keyBindingList));
     }
 
+    void Actions::KeyChordEditor_PreviewKeyDown(Windows::Foundation::IInspectable const& sender, Windows::UI::Xaml::Input::KeyRoutedEventArgs const& e)
+    {
+        const auto& senderTB{ sender.as<TextBox>() };
+        const auto& kbdVM{ senderTB.DataContext().as<Editor::KeyBindingViewModel>() };
+        if (e.OriginalKey() == VirtualKey::Enter)
+        {
+            // Fun fact: this is happening _before_ "_ProposedKeys" gets updated
+            // with the two-way data binding. So we need to directly extract the text
+            // and tell the view model to update itself.
+            get_self<KeyBindingViewModel>(kbdVM)->AttemptAcceptChanges(senderTB.Text());
+
+            // For an unknown reason, when 'AcceptChangesFlyout' is set in the code above,
+            // the flyout isn't shown, forcing the 'Enter' key to do nothing.
+            // To get around this, detect if the flyout was set, and display it
+            // on the text box.

```

Comment:
wait, we can't work around this by just adding a 

```c++
kbdVM.AcceptChangesFlyout().ShowAt(senderTB);
```

to line 281?

I suppose that line doesn't know which `TextBox` raised the event, so that's fine. Weird, but fine.

Reply:
Yeah, we need a reference to `senderTB`. 😭 We're really relying heavily on MVVM here haha